File: backend/domains/gateway/presentation/routers/provider_profiles.py
"""GET /api/v1/gateway/provider-profiles — 上游方案 SSOT。"""

from __future__ import annotations

from fastapi import APIRouter

from domains.gateway.domain.upstream_profile import UpstreamProtocol
from domains.gateway.domain.upstream_profile_registry import list_all_upstream_profiles
from domains.gateway.presentation.schemas.provider_profiles import (
    ProviderProfileApiBaseResponse,
    ProviderProfileResponse,
    ProviderProfilesListResponse,
)
from domains.identity.presentation.deps import RequiredAuthUser

router = APIRouter()


@router.get("/provider-profiles", response_model=ProviderProfilesListResponse)
async def list_provider_profiles(
    _user: RequiredAuthUser,
) -> ProviderProfilesListResponse:
    profiles = [
        ProviderProfileResponse(
            id=p.id,
            provider=p.provider,
            label=p.label,
            api_bases=ProviderProfileApiBaseResponse(
                openai_compat=p.api_bases.get(UpstreamProtocol.OPENAI_COMPAT),
                anthropic_native=p.api_bases.get(UpstreamProtocol.ANTHROPIC_NATIVE),
            ),
            models_list_path=p.models_list_path,
            default_call_shape=p.default_call_shape.value,
            probe_supported=p.probe_supported,
        )
        for p in list_all_upstream_profiles()
    ]
    return ProviderProfilesListResponse(profiles=profiles)


__all__ = ["router"]

File: frontend/src/api/gateway/provider-profiles.ts
/**
 * GET /api/v1/gateway/provider-profiles — 上游方案 SSOT
 */

import { apiClient } from '@/api/client'

import { GATEWAY_API_BASE } from './_base'

export interface ProviderProfileApiBases {
  openai_compat: string | null
  anthropic_native: string | null
}

export interface ProviderProfileItem {
  id: string
  provider: string
  label: string
  api_bases: ProviderProfileApiBases
  models_list_path: string
  default_call_shape: string
  probe_strategy: string
  probe_protocol: string
  probe_supported: boolean
}

export interface ProviderProfilesListResponse {
  profiles: ProviderProfileItem[]
}

export const providerProfilesApi = {
  listProviderProfiles: async (): Promise<ProviderProfilesListResponse> => {
    return apiClient.get<ProviderProfilesListResponse>(`${GATEWAY_API_BASE}/provider-profiles`)
  },
}
